### PR TITLE
chore(deps)!: bump k8s dependencies to 1.30

### PR DESCRIPTION
**Pull request by blkperl**
### Motivation

* K8s 1.26 is no longer receiving security patches

### Modifications

* updated `go.mod` to k8s 1.30
* updated `dist/kubernetes.swagger.json` version to 1.30
* updated `workflow/executor/resource.go` to non deprecated functions to appease the linter
* updated from controller-gen@v0.14.0 to controller-gen@v0.15.0
* updated various files to the new ptr library to appease the linter
* changed the metrics test contains assertion from status_code=200 to status_code=400 to pass the test
* changed `getEvents` test helper to truncate annotations from events so that `TestEventNodeEvents` passes

### Verification

Existing test coverage

### CODE REVIEW

**Comment by blkperl** [comment]
@agilgur5 is the mysql test a known failure or did the PR somehow break the metric?

**Comment by agilgur5** [comment]
I don't recall this as a prior flake. You can re-run individual failed tests now (#13000) with a `/retest` comment to verify a flake. See my below comment as an example

**Comment by blkperl** [comment]
hey @terrytangyuan could you take a look please? It's one PR like you requested in https://github.com/argoproj/argo-workflows/pull/12865

File: Makefile
Comment (terrytangyuan) [inline Makefile:296]: What package is this?

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
Hold for v3.6

**Comment by terrytangyuan** [comment]
Thank you for this! Since this is a breaking change, we cannot merge now.

File: Makefile
Comment (blkperl) [reply Makefile:296]: I'm not sure what you mean. It fails to run go install when I bump the version.

```
go install k8s.io/code-generator/cmd/go-to-protobuf@v0.30.3
```

**Comment by Joibel** [comment]
@terrytangyuan, any reason not to merge this to `main` only, which will become 3.6?

**Comment by terrytangyuan** [comment]
You are right. We should merge it to main. Do you want to do another review? 

@blkperl Could you resolve the conflicts? Thanks!

**Comment by blkperl** [comment]
@terrytangyuan I've resolved the conflicts. Thanks!

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**
